### PR TITLE
TandemFoil: AdamW vs Lion optimizer comparison at golden LR settings

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The TandemFoil golden config uses Lion optimizer at lr=1.25e-4. We have not tested whether AdamW can match or beat Lion on this dataset. AirfRANS uses AdamW successfully, and the optimal optimizer choice is dataset-dependent. The ~0.1x LR rule of thumb suggests AdamW lr~1.25e-3 is the AdamW equivalent, but we should also try lr=7e-4 (the AirfRANS AdamW golden) and lr=2e-4 (conservative). This comparison will settle the optimizer question for TandemFoil definitively.

## Instructions
Test AdamW at three learning rates against Lion baseline, keeping all other golden settings (3L/192d, gc=1.0, WD=1e-2, T_max=10).

**Run 1 (CUDA 0) — AdamW lr=1.25e-4 (direct LR swap, same value as Lion):**
```bash
CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset tandemfoil \
  --model-layers 3 \
  --model-hidden-dim 192 \
  --optimizer adamw \
  --lr 1.25e-4 \
  --weight-decay 1e-2 \
  --grad-clip 1.0 \
  --cosine-t-max 10 \
  --model-slices 64 \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --wandb-group tandem-adamw-vs-lion \
  --wandb-name tandem-adamw-lr125e4
```

**Run 2 (CUDA 1) — AdamW lr=7e-4 (AirfRANS golden AdamW LR):**
```bash
CUDA_VISIBLE_DEVICES=1 python train.py \
  --dataset tandemfoil \
  --model-layers 3 \
  --model-hidden-dim 192 \
  --optimizer adamw \
  --lr 7e-4 \
  --weight-decay 1e-2 \
  --grad-clip 1.0 \
  --cosine-t-max 10 \
  --model-slices 64 \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --wandb-group tandem-adamw-vs-lion \
  --wandb-name tandem-adamw-lr7e4
```

**Run 3 (CUDA 2) — AdamW lr=2e-4 (conservative, between the two):**
```bash
CUDA_VISIBLE_DEVICES=2 python train.py \
  --dataset tandemfoil \
  --model-layers 3 \
  --model-hidden-dim 192 \
  --optimizer adamw \
  --lr 2e-4 \
  --weight-decay 1e-2 \
  --grad-clip 1.0 \
  --cosine-t-max 10 \
  --model-slices 64 \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --wandb-group tandem-adamw-vs-lion \
  --wandb-name tandem-adamw-lr2e4
```

## Baseline
- Dataset: TandemFoil
- Primary metric: `val_primary/surface_pressure_mae` (lower is better)
- Current best: **30.10** (PR #2810, 3L/192d + Lion lr=1.25e-4 + gc=1.0 + WD=1e-2 + T_max=10)

## Expected Outcome
Report `val_primary/surface_pressure_mae` for each run. If any AdamW variant beats 30.10, we switch the TandemFoil golden config to AdamW. The best AdamW LR also tells us the optimal scale factor for future Lion→AdamW conversions on this dataset. If Lion still wins, this confirms the optimizer choice and lets us focus on other levers.

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing this PR because the assigned student pod was retired during the DrivAerML and cross-dataset refocus on 2026-04-21. We are freeing capacity for the new relaunch wave, and this lane is no longer being actively pursued. Any useful results already logged here or in W&B remain part of the research record.
